Class/Tools.php

```diff
@@ -400,24 +529,28 @@ public static function htmlTextarea($name, $content, $cols = 60, $rows = 5, $opt
         return self::htmlTextInput($name, $label, $content, $options);
     }
 
-    // HTML notation for the <input> tag. See html_textinput() for more info.
+    /**
+     * HTML notation for the <input> tag. See self::htmlTextinput() for more info.
+     */
     public static function htmlInput($name, $label, $value, $options = array())
     {
         return self::htmlTextInput($name, $label, $value, $options);
     }
 
-    /** HTML notation for the <input> or <textarea> tag
-     * @param string name
-     * @param string label, omitted if empty, translated if true
-     * @param mixed value, either given directly or as an array in the [name] index
-     * @param array options (if a string -> assume "type" attribute)
+    /**
+     * HTML notation for the <input> or <textarea> tag. Used by self::htmlInput() and self::htmlTextarea()
+     *
+     * @param string $name element name 
+     * @param string $label label, omitted if empty, translated if true
+     * @param mixed $value value, either given directly or as an array in the [name] index
+     * @param mixed $options (optional) options. Either an associative array or string containing type
      *     [type] - input's type, 'text' by default
      *     [before], [between], [after] - HTML to insert before/between/after <label> and <input>
      *     [table] - refill [before], [between], [after] to make a table row
-     *     [flag] - <code> -> prefix the label with a flag icon of given code
      *     [random-id] - append random number to the id attribute
      *     [label-after] - <tag> goes first, <label> after
      *     [label-html] - label given as raw HTML, don't escape HTML entities
+     *     [label-class] - class(es) for label
      *     other tag's attributes - will be specified (except for NULLs)
      */
     protected static function htmlTextInput($name, $label, $value, $options = array())
```

Comment:
A když níže $value['name'] nebude definovaný a chybu to nevyhodí, tak $value bude obsahovat co?